google-cloud-compute-v1/samples/quickstart.rb

```diff
@@ -0,0 +1,192 @@
+    puts "Instance #{instance_name} created."
+    instance
+  rescue ::Google::Cloud::Error => e
+    warn "Exception during creation:", e
+  end
+end
+
+# [END compute_instances_create]
+
+# [START compute_instances_list]
+
+# Lists all instances in the given zone in the specified project.
+#
+# @param [String] project project ID or project number of the Cloud project you want to use.
+# @param [String] zone name of the zone you want to use. For example: “us-west3-b”
+# @return Array of instances.
+def list_instances project:, zone:
+  client = ::Google::Cloud::Compute::V1::Instances::Rest::Client.new
+  instance_list = client.list project: project, zone: zone
+
+  puts "Instances found in zone #{zone}:"
+  instance_list.items.each do |instance|
+    puts " - #{instance.name} (#{instance.machine_type})"
+  end
+  instance_list.items
+end
+
+# [END compute_instances_list]
+
+# [START compute_instances_list_all]
+# Returns a dictionary of all instances present in a project, grouped by their zone.
+#
+# @param [String] project project ID or project number of the Cloud project you want to use.
+# @return A hash with zone names as keys (in form of "zones/{zone_name}") and
+#   arrays of instances as values.
+def list_all_instances project:
+  client = ::Google::Cloud::Compute::V1::Instances::Rest::Client.new
+  agg_list = client.aggregated_list project: project
+  all_instances = {}
+  puts "Instances found:"
+  agg_list.items.each do |zone, list|
+    next if list.instances.empty?
+    all_instances[zone] = list.instances
+    puts " #{zone}:"
+    list.instances.each do |instance|
+      puts " - #{instance.name} (#{instance.machine_type})"
+    end
+  end
+  all_instances
+end
+# [END compute_instances_list_all]
+
+# [START compute_instances_delete]
+
+# Sends an instance deletion request to the Compute Engine API and waits for it to complete.
+#
+# @param [String] project project ID or project number of the Cloud project you want to use.
+# @param [String] zone name of the zone you want to use. For example: “us-west3-b”
+# @param [String] instance_name name of the instance you want to delete.
+def delete_instance project:, zone:, instance_name:
+  client = ::Google::Cloud::Compute::V1::Instances::Rest::Client.new
+
+  puts "Deleting #{instance_name} from #{zone}..."
+  begin
+    operation = client.delete project: project, zone: zone, instance: instance_name
+    operation_client = ::Google::Cloud::Compute::V1::ZoneOperations::Rest::Client.new do |config|
+      # Set the timeout to 120 seconds as delete requests tend take longer than the default
+      # Net::HTTP timeout of 60 seconds.
+      config.timeout = 120
+    end
+    operation = operation_client.wait operation: operation.name, project: project,
+                                      zone: zone while operation.status == :RUNNING
+    warn "Error during deletion:", operation.error unless operation.error.nil?
```

Comment:
In case of error we should probably throw or at least not output `Instance deleted` below, it might not be.